src/RequestLogger.Tests/Unit/ProgramTests.cs

```diff
@@ -0,0 +1,200 @@
+using System.Text.Json;
+using FakeItEasy;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Repository;
+using RequestLogger.Dto;
+using RequestLogger.Services;
+using RequestLogger.Services.Interfaces;
+using AutoFixture;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage;
+using RequestLogger.Tests.Helpers;
+
+#pragma warning disable CS8620
+
+namespace RequestLogger.Tests.Unit;
+
+public class ProgramTests
+{
+    private readonly WebApplicationFactory<Program> _sut;
+    private readonly Fixture _fixture;
+
+    public ProgramTests()
+    {
+        var root = new InMemoryDatabaseRoot();
+        
+        _sut = new RequestLoggerAppBuilderFactory<Program>()
+            .WithWebHostBuilder(builder => builder.ConfigureServices(services =>
+            {
+                var descriptor = services.SingleOrDefault(
+                    d => d.ServiceType ==
+                         typeof(DbContextOptions<RequestLoggerContext>));
+                
+                services.Remove(descriptor!);
+                services.AddScoped<IRequestLoggerService, RequestLoggerService>();
+                services.AddScoped<IBlacklistService, BlacklistService>();

```

Comment:
These 2 `.AddScoped<>` requests shouldn't be required as they're already configured in `Program.cs`